src/dataAssimilation/FieldIO.cpp

```diff
@@ -0,0 +1,312 @@
+    header.append(fmt::format("###XML;{}\n", xml_filename));
+    return header;
+}
+
+void FieldIO::read_field(std::ifstream &file_stream, Field &field) {
+    std::string line;
+    getline(file_stream, line);
+    std::vector<std::string> divided_string = Utility::split(line, ';');
+    size_t counter = 0;
+    for (const std::string &part: divided_string) {
+        field.data[counter] = std::stod(part);
+        counter++;
+    }
+}
+
+void FieldIO::read_fields(const Settings::data_assimilation::field_changes &field_changes,
+                          Field &u, Field &v, Field &w,
+                          Field &p, Field &T, Field &C) {
+    std::string line;
+    if (!field_changes.changed) {
+        return;
+    }
+    // no changes -> read original file
+    std::ifstream file_changes(field_changes.filename, std::ifstream::binary);
+    if (file_changes.is_open()) {  // could not open file -> read original file + warning
+        if (field_changes.u_changed) {
+            read_field(file_changes, u);
+            m_logger->debug("read changed u Field");
+            getline(file_changes, line);
+        }
+        if (field_changes.v_changed) {
+            read_field(file_changes, v);
+            m_logger->debug("read changed v Field");
+        }
+        if (field_changes.w_changed) {
+            read_field(file_changes, w);
+            m_logger->debug("read changed w Field");
+        }
+        if (field_changes.p_changed) {
+            read_field(file_changes, p);
+            m_logger->debug("read changed p Field");
+        }
+        if (field_changes.T_changed) {
+            read_field(file_changes, T);
+            m_logger->debug("read changed T Field");
+        }
+        if (field_changes.C_changed) {
+            read_field(file_changes, C);
+            m_logger->debug("read changed C Field");
+        }
+    } else {
+        m_logger->warn(fmt::format("File '{}' could not be opened. No changes will be applied.", field_changes.filename));
+    }
+}
+
+void FieldIO::read_fields(const real t_cur,
+                          const Settings::data_assimilation::field_changes &field_changes,
+                          Field &u, Field &v, Field &w,
+                          Field &p, Field &T, Field &C) {
+    std::ifstream file_original(m_filename, std::ifstream::binary);
+    size_t n = static_cast<size_t>(std::round(t_cur / DomainData::getInstance()->get_physical_parameters().dt)) - 1;
+    file_original.seekg(m_positions[n]);
+
+    std::string line;
+    getline(file_original, line);
+    m_logger->debug("read time step {}", line);
+
+    if (field_changes.changed) {  // no changes -> read original file
+        std::ifstream file_changes(field_changes.filename, std::ifstream::binary);
+        if (file_changes.is_open()) {  // could not open file -> read original file + warning
+            if (field_changes.u_changed) {
+                read_field(file_changes, u);
+                getline(file_original, line);
```

Comment:
what about the result? If you only want the set the current stream position indicator use maybe `ifstream::seekg`